main.go

```diff
@@ -232,6 +232,15 @@ func (h k8sHelpers) GetNode(cli *client.Client) (*api.Node, error) {
 }
 
 func (h k8sHelpers) AddLabels(n *api.Node, labels Labels) {
+	// Remove nfd node labels that no longer apply
+	for k := range n.Labels {
+		// If the label is not in the labels we are adding
+		// and it is an nfd label, delete it.
+		if _, ok := labels[k]; !ok && strings.Contains(k, Namespace) {
```

Comment:
s/Namespace/prefix/. See https://github.com/kubernetes-incubator/node-feature-discovery/pull/28#discussion_r85028520 for context.